twitch.go

```diff
@@ -110,31 +115,31 @@ func verifyWebhook(r *http.Request, requestBody []byte, hmacKeys [][]byte) bool
 		hasher = sha512.New
 	default:
 		log.Println("unknown signature algorithm", method)
-		return false
+		return nil, false
 	}
 
 	timestamp, err := getCoolHeader("Twitch-Eventsub-Message-Timestamp", r)
 	if err != nil {
 		log.Println(err)
-		return false
+		return nil, false
 	}
 
 	msgId, err := getCoolHeader("Twitch-Eventsub-Message-Id", r)
 	if err != nil {
 		log.Println(err)
-		return false
+		return nil, false
 	}
 	for _, hmacKey := range hmacKeys {
-		calculatedHMAC := hmac.New(hasher, hmacKey)
+		calculatedHMAC := hmac.New(hasher, hmacKey.secret)
 		calculatedHMAC.Write([]byte(msgId))
 		calculatedHMAC.Write([]byte(timestamp))
 		calculatedHMAC.Write(requestBody)
 		expectedMAC := calculatedHMAC.Sum(nil)
 		if hmac.Equal(expectedMAC, signature) {
-			return true
+			return hmacKey.permissions, true
```

Comment:
thoughts on just returning allowed permissions here, rather than the bool? it seems superfluous